### PR TITLE
Figure.savefig: Support saving figures in PPM (.ppm) format

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Support PPM format and also sort formats alphabetically.

**Preview**: https://pygmt-dev--2771.org.readthedocs.build/en/2771/api/generated/pygmt.Figure.savefig.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: pygmt/figure.py [hunk 1]
```diff
@@ -263,6 +263,7 @@ def savefig(
         - JPEG (``.jpg`` or ``.jpeg``)
         - PDF (``.pdf``)
         - BMP (``.bmp``)
+        - PPM (``.ppm``)
```
Comment (yvonnefroehlich) [inline pygmt/figure.py:266]: Maybe it make sense to also sort this list alphabetically?

Comment (seisman) [reply pygmt/figure.py:269]: Thanks. Done in 473bef8f4354b181c0a3386b26d5b48989bc7854.

File: pygmt/figure.py [hunk 2]
```diff
@@ -259,14 +259,15 @@ def savefig(
...
         - BMP (``.bmp``)
-        - TIFF (``.tif``)
-        - GeoTIFF (``.tiff``)
         - EPS (``.eps``)
+        - GeoTIFF (``.tiff``)
+        - JPEG (``.jpg`` or ``.jpeg``)
         - KML (``.kml``)
+        - PDF (``.pdf``)
+        - PNG (``.png``)
+        - PPM (``.ppm``)
```
Comment (yvonnefroehlich) [inline pygmt/figure.py:269]: I am not really familiar with the `ppm` format, but maybe we should add the `ppm` format also to the list of raster images for which the `anit_alias` parameter can be used (see line 287):

```
anti_alias: bool
            If ``True``, will use anti-aliasing when creating raster images
            (BMP, PNG, JPEG, TIFF, and GeoTIFF). More specifically, it passes
            the arguments ``"t2"`` and ``"g2"`` to the ``anti_aliasing``
            parameter of :meth:`pygmt.Figure.psconvert`. Ignored if creating
            vector graphics.
```

Comment (seisman) [reply pygmt/figure.py:268]: In 0c8184710ea93a59ce6e3b2027114edb53e63421, I've revised the description of supported images formats and clarify the raster and vectors images that PyGMT supports.

Comment (seisman) [reply pygmt/figure.py:268]: @yvonnefroehlich OK to approve?

**Review by yvonnefroehlich — APPROVED** [review]

File: pygmt/figure.py (hunk 2 above)
Comment (yvonnefroehlich) [reply pygmt/figure.py:268]: Looks good to me!

**Pull request merged**